src/main/java/gov/usgs/earthquake/distribution/WebSocketClient.java

```diff
@@ -10,121 +10,105 @@
 @ClientEndpoint
 public class WebSocketClient implements Runnable {
 
-  public static final long DEFAULT_TIMEOUT_MILLIS = 30000;
+  public static final long DEFAULT_RECONNECT_INTERVAL = 30000;
 
   private Session session;
   private URI endpoint;
   private WebSocketListener listener;
   private Thread thread;
-  private long timeoutMillis;
+  private long reconnectInterval;
   private boolean stopThread;
-  private boolean doReconnect;
-  private Object sync = new Object();
+  private boolean reconnect;
+  private Object close = new Object();
 
   public WebSocketClient (URI endpoint, WebSocketListener listener) {
-    this (endpoint, listener, DEFAULT_TIMEOUT_MILLIS);
+    this (endpoint, listener, true, DEFAULT_RECONNECT_INTERVAL);
   }
 
-  public WebSocketClient(URI endpoint, WebSocketListener listener, long timeoutMillis) {
+  public WebSocketClient(URI endpoint, WebSocketListener listener, boolean reconnect, long reconnectInterval) {
     this.listener = listener;
     this.endpoint = endpoint;
-    this.timeoutMillis = timeoutMillis;
+    this.reconnect = reconnect;
+    this.reconnectInterval = reconnectInterval;
   }
 
 
   public void startup() {
     stopThread = false;
-    this.thread = new Thread(this);
-    this.thread.start();
+    thread = new Thread(this);
+    thread.start();
   }
 
   public void shutdown() throws Exception {
     // bring down thread
     stopThread = true;
-    thread.interrupt();
+    synchronized (close) {
+      thread.interrupt();
+    }
     thread.join();
+    thread = null;
+  }
+
+  public void connect() throws Exception {
+    // try to connect to server
```

Comment:
Not expected, but check isConnected and return if already connected.

Reply:
added this check